### PR TITLE
Add Presets for opentelemetry-kube-stack

### LINKED ISSUE #1081: [opentelemetry-kube-stack] Generate Quickstart Configuration
### original issue
#562

### context
https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1058#issuecomment-1995933786

### description
Make it easy for a user to get started with all of the features OpenTelemetry offers in a kubernetes environment by creating various collectors to solve monitoring issues.

### PR DESCRIPTION
**Pull request by jaronoff97**
Closes #1081 

This PR adds in full presets and default collectors for the chart.

TODO:
* ~generate examples~
* ~add some better validations~
  * This will be done in a followup https://github.com/open-telemetry/opentelemetry-helm-charts/issues/1191

### CODE REVIEW

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:12]: Should we define the daemon as a known "collector" in the base values.yaml that just exports to debug and then a user simply says "here's the exporter I want"... OR do we give them an empty collectors map and then use these examples for how we expect users to apply? I think it's simpler to do the former as it gives users a one click install to begin running a good set of collectors

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:10]: at first i liked the idea of providing an OTLP exporter preset, but now i'm not sure... interested in other peoples thoughts!

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:4]: Yes I know I need to document this values file and what we expect for the output 😓

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:20]: this technically could go in the defaultCRConfig, but I wanted to show that users can override these individually

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:30]: I'm not 100% sure doing this works for the SDKs... I haven't tested this but i think it would work!

File: charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml
Comment (TylerHelmuth) [reply charts/opentelemetry-kube-stack/examples/cloud-demo/values.yaml:10]: I think no.  Configuring OTLP exporters for your favorite backend are going to vary to widely to make this useful I think.

**Comment by jaronoff97** [comment]
blocked by https://github.com/open-telemetry/opentelemetry-helm-charts/pull/1175#pullrequestreview-2046366002

File: charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/bridge.yaml
Comment (jvoravong) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/bridge.yaml:42]: Might suggest removing the OTEL_* prefix from OTEL_K8S_NODE_NAME, OTEL_K8S_NAMESPACE, and OTEL_K8S_POD_IP for simplification unless we have reason to keep the prefix.

File: charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/bridge.yaml
Comment (jvoravong) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/bridge.yaml:11]: Extra new line, would fix in the the bridge template

File: charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/clusterrole.yaml
Comment (jvoravong) [inline charts/opentelemetry-kube-stack/examples/cloud-demo/rendered/clusterrole.yaml:50]:
```suggestion
```

File: charts/opentelemetry-kube-stack/templates/_config.tpl
Comment (jvoravong) [inline charts/opentelemetry-kube-stack/templates/_config.tpl:295]: I think we need to add a storage configuration for offset tracking.
```suggestion
    start_at: beginning
    storage: file_storage
```

**Comment by jvoravong** [comment]
Most of these changes lgtm. Left a couple small comments.

**Comment by TylerHelmuth** [comment]
@jaronoff97 FYI I'm not planning to look at this until you remove the `do-not-merge` tag

**Comment by jaronoff97** [comment]
@TylerHelmuth yep, and im not going to remove that until #1176 is done :D

File: charts/opentelemetry-kube-stack/values.yaml
Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml:403]: I was thinking this would be a useful thing that most users would want to add in... I can remove some of the proxying and instead just set some default values for this chart

File: charts/opentelemetry-kube-stack/crds/crd-opentelemetry.io_opampbridges.yaml
Comment (jaronoff97) [inline charts/opentelemetry-kube-stack/crds/crd-opentelemetry.io_opampbridges.yaml:1]: doing it this way is my only option right now, until #677 is complete

**Comment by jaronoff97** [comment]
I need to rebase off main and use the new CRDs :D i'm on vacation right now, but I want to get to this soon.

File: charts/opentelemetry-kube-stack/values.yaml
Comment (TylerHelmuth) [reply charts/opentelemetry-kube-stack/values.yaml:417]: setting the batchprocessor as a default processor feels better than a preset for it.  All the presets exist to handle both component and k8s setup, and the batch processor has no k8s setup.

File: charts/opentelemetry-kube-stack/values.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/values.yaml:431]: should we use the chart name/release name as part of these collectors?

File: charts/opentelemetry-kube-stack/values.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/values.yaml:444]: Is this a TA thing?

File: charts/opentelemetry-kube-stack/values.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/values.yaml:448]: a daemonset shouldn't be used to collect events, I believe it will result in duplicate data. Should be used in the cluster instance.

File: charts/opentelemetry-kube-stack/templates/_config.tpl
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/templates/_config.tpl:1]: I'm pretty confused by this file.  Is it handling the collector config like from the collector chart? I dont understand the `scrape_configs_file` stuff.

File: charts/opentelemetry-kube-stack/templates/_config.tpl
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/templates/_config.tpl:297]: this can use the container-logs parser now, the collector chart has been updated

File: charts/opentelemetry-kube-stack/values.yaml
Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml:431]: that's automatically prefixed, this is the suffix... maybe i should change this to be "suffix" and then expose a fullname override

File: charts/opentelemetry-kube-stack/values.yaml
Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml:444]: this is a custom capability of my devising to allow the collector to scrape cadvisor metrics which are not available anywhere else right now.

Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml:444]: it also makes it easy for people to add a scrape config if they want one

File: charts/opentelemetry-kube-stack/templates/_config.tpl
Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/templates/_config.tpl:1]: yeah it's very similar to the collector's config stuff. The point of the scrape_configs_file is to get prometheus metrics from kubernetes components that have no otel equivalent today (cadvisor, node exporter, and then scraping pods on node that have available targets). I can document this more

File: charts/opentelemetry-kube-stack/values.yaml
Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml]: probably not, i removed it.

File: charts/opentelemetry-kube-stack/values.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/values.yaml:474]: Can we rely on the processor's default values and change this to `batch: {}`?

Comment (jaronoff97) [reply charts/opentelemetry-kube-stack/values.yaml:474]: I try to be explicit where possible, IMO the default batch size being 8k is too large

**Review by TylerHelmuth — APPROVED** [review]

File: charts/opentelemetry-kube-stack/Chart.yaml
Comment (TylerHelmuth) [inline charts/opentelemetry-kube-stack/Chart.yaml:16]:
```suggestion
appVersion: 0.103.1
```

Comment (TylerHelmuth) [reply charts/opentelemetry-kube-stack/Chart.yaml:16]: Missed that this was the operator

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**
